### PR TITLE
Integrating Google Secrets Manager with CI/CD (SCP-5659)

### PR DESCRIPTION
**Pull request by bistline**
#### BACKGROUND & CHANGES
This update fully integrates Google Secret Manager into our normal CI/CD pipeline.  Now, instead of loading secrets/configurations from vault, they are pulled from GSM.  Additionally, `gcloud` usage in workflows now uses the `google-github-actions/setup-gcloud` action, rather than the previous Docker container implementation.  This reduces the code complexity significantly while not affecting performance.

There will be future work to further refine this pipeline such that secrets are not directly loaded from GSM in a workflow, and rather everything is managed by Terraform, and any updates in GSM are propagated over to Github secrets.  This would improve masking/security while not affecting functionality.  Since this is out-of-scope for this PR, this work has been tracked in SCP-5669, and is be scheduled for Q3 (after DevOps finishes the bulk of their FedRAMP migration).

#### MANUAL TESTING
The automated CI runs below demonstrate that this is working.  Unfortunately, the Docker build & deployment workflows cannot be tested until this has merged due to known limitations with Github Actions workflows.  Any regressions found during this process will have to be dealt with in another PR, though I don't anticipate there being any problems.

### CODE REVIEW

**Comment by bistline** [comment]
Build failure is known transient issue with HCA metadata exports.

File: NON_CONTAINERIZED_DEV_README.md [hunk 1]
```diff
@@ -83,8 +82,8 @@ ulimit -Sn 10240
 
 ## REGULAR DEVELOPMENT
 Adding `source <<path-to-single-cell-portal-core>>/config/secrets/.source_env.bash` to your .zschrc or .bash_profile will source the 
-secrets read from Vault to each new shell, saving you the trouble of rerunning the setup process every time you open a 
-new shell.  
+secrets read from Google Secrets Manager (GSM) via `gcloud` to each new shell, saving you the trouble of rerunning the
```
Comment (eweitz) [inline NON_CONTAINERIZED_DEV_README.md:85]: Thanks for updating the docs, as well as the code!

**Review by eweitz — APPROVED** [review]
Code looks good!  Thanks for the thorough replacement in this repo.

I know we still have references to Vault in:
* [Ingest Pipeline](https://github.com/broadinstitute/scp-ingest-pipeline)
* Internal documents: [SCP playbook](https://docs.google.com/document/d/1vJz6hXdjAu8LBAfIS4oMy7yLIUX177gU1H7BxTPPzAU/edit) and [onboarding checklist](https://docs.google.com/document/d/1lRYQB3zQB2iHpdE8LFu88a-b74qnIm0VUZTMFv4eOXM/edit#bookmark=id.wjloykr2lb6h)

But at a glance this PR looks like it'll remove all references to Vault in SCP Core.

**Review by jlchang — APPROVED** [review]
Looking forward to being off Vault well before it is deprecated. Thanks so much for doing this ahead of time!

**Pull request merged**